### PR TITLE
Browser login fix + Logout async

### PR DESCRIPTION
**Pull request by shrishrirang**

### CODE REVIEW

**Comment by shrishrirang** [comment]
\+ @danderson00

File: sdk/src/MobileServiceClient.js [hunk 1]
```diff
@@ -6,11 +6,12 @@
 /// <reference path="C:\Program Files (x86)\Microsoft SDKs\Windows\v8.0\ExtensionSDKs\Microsoft.WinJS.1.0\1.0\DesignTime\CommonConfiguration\Neutral\Microsoft.WinJS.1.0\js\ui.js" />
 /// <reference path="Generated\MobileServices.DevIntellisense.js" />
 
-var _ = require('./Utilities/Extensions');
-var Validate = require('./Utilities/Validate');
-var Platform = require('Platforms/Platform');
-var MobileServiceTable = require('./MobileServiceTable').MobileServiceTable;
-var MobileServiceLogin = require('./MobileServiceLogin').MobileServiceLogin;
+var _ = require('./Utilities/Extensions'),
+    Promises = require('./Utilities/Promises'),
```
Comment (danderson00) [inline sdk/src/MobileServiceClient.js]: Promises is unused. Are we linting the JS client?

File: sdk/src/MobileServiceClient.js [hunk 2]
```diff
@@ -357,12 +358,17 @@ MobileServiceClient.prototype.login = Platform.async(
...
     /// Log a user out of a Mobile Services application.
+    /// <param name="callback" type="Function" mayBeNull="true">
+    /// Optional callback accepting error as a parameter.
+    /// </param>
     /// </summary>
+    
     this.currentUser = null;
-};
+    callback();
+});
```
Comment (danderson00) [inline sdk/src/MobileServiceClient.js]: You will have to fill me in on why we are doing this.

File: sdk/test/winJS/tests/unit/mobileServicesClient.js [hunk 3]
```diff
@@ -318,12 +318,13 @@ $testGroup('MobileServiceClient.js',
 
     $test('logout')
     .description('Verify Authentication.logout undoes the effects of logging in')
```
Comment (danderson00) [inline sdk/test/winJS/tests/unit/mobileServicesClient.js]: This is not a very good test name. It's not entirely true (the token does not get expired) and it's too general. Should describe what it tests, i.e. "clears current user" or similar

File: sdk/src/MobileServiceClient.js (hunk 2 above)
Comment (shrishrirang) [inline sdk/src/MobileServiceClient.js]: As discussed, this is a forward looking change as I expect we might need to invalidate the token in the future to do a real logout.

File: sdk/src/MobileServiceClient.js (hunk 1 above)
Comment (shrishrirang) [inline sdk/src/MobileServiceClient.js]: Good point. The default behavior of grunt-jshint is to ignore unused variable errors and probably many other common errors. The SDK build script does specify a list of errors that jshint should look for. I am going to create an issue to fix jshint. Quickly turning the unused variable error flag listed a bunch of errors!! As part of this commit, I am just removing 'Promises'

File: sdk/test/winJS/tests/unit/mobileServicesClient.js (hunk 3 above)
Comment (shrishrirang) [inline sdk/test/winJS/tests/unit/mobileServicesClient.js]: Fixed.

**Comment by danderson00** [comment]
Looks good!

**Pull request merged**